File: data-ts/src/apy_msol.ts
const BASE_URL = "https://api.marinade.finance";

export async function getMsolApy(period: string, time?: string) {
  const url = new URL(`${BASE_URL}/msol/apy/${period}`);
  if (time) {
    url.searchParams.append("time", time);
  }

  const res = await fetch(url.toString(), {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
    },
  });

  if (!res.ok) {
    throw new Error(`Failed to fetch APY: ${res.status} ${res.statusText}`);
  }

  return await res.json();
}

//example usage 
(async () => {
  try {
    const data = await getMsolApy("7d");   
    console.log("mSOL APY data:", data);
    console.log("mSOL APY:", data.value * 100 + "%");
  } catch (err) {
    console.error(err);
  }
})();

File: data_plane/apy_msol.py
import requests

BASE_URL = "https://api.marinade.finance"

def get_msol_apy(period: str, time: str = None):
    url = f"{BASE_URL}/msol/apy/{period}"
    params = {}
    if time:
        params["time"] = time

    response = requests.get(url, params=params)
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":
    print(get_msol_apy("7d")) 
